fonduer/parser/visual.py

```diff
@@ -178,19 +185,23 @@ def get_anchors(l, u):
             return l, u, L, U
 
         def display_match_counts():
-            matches = sum([
-                html_to_pdf[i] is not None and self.html_word_list[i][1] ==
-                self.pdf_word_list[html_to_pdf[i]][1]
-                for i in range(len(self.html_word_list))
-            ])
+            matches = sum(
+                [
+                    html_to_pdf[i] is not None
+                    and self.html_word_list[i][1]
+                    == self.pdf_word_list[html_to_pdf[i]][1]
+                    for i in range(len(self.html_word_list))
+                ]
+            )
             total = len(self.html_word_list)
-            self.logger.info("({:d}/{:d}) = {:.2f}".format(
-                matches, total, matches / total))
+            self.logger.info(
+                "({:d}/{:d}) = {:.2f}".format(matches, total, matches / total)
+            )
             return matches
 
         N = len(self.html_word_list)
         M = len(self.pdf_word_list)
-        assert (N > 0 and M > 0)
+        assert N > 0 and M > 0
```

Comment:
use logging

Reply:
Logging and assertions serve different purposes. Checking and throwing an exception would be a better fix if you don't like the assertion.